src/Microsoft.CodeAnalysis.Testing/README.md

```diff
@@ -17,52 +17,48 @@ To reference prerelease packages, add a **NuGet.Config** file to your solution d
-  * Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.MSTest
-  * Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing.MSTest
-  * Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing.MSTest
+  * Microsoft.CodeAnalysis.CSharp.Analyzer.Testing
+  * Microsoft.CodeAnalysis.CSharp.CodeFix.Testing
+  * Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing
+  * Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing
 * Visual Basic
-  * Microsoft.CodeAnalysis.VisualBasic.Analyzer.Testing.MSTest
-  * Microsoft.CodeAnalysis.VisualBasic.CodeFix.Testing.MSTest
-  * Microsoft.CodeAnalysis.VisualBasic.CodeRefactoring.Testing.MSTest
-  * Microsoft.CodeAnalysis.VisualBasic.SourceGenerators.Testing.MSTest
+  * Microsoft.CodeAnalysis.VisualBasic.Analyzer.Testing
+  * Microsoft.CodeAnalysis.VisualBasic.CodeFix.Testing
+  * Microsoft.CodeAnalysis.VisualBasic.CodeRefactoring.Testing
+  * Microsoft.CodeAnalysis.VisualBasic.SourceGenerators.Testing
 
-### NUnit
+### Obsolete Packages
 
-* C#
-  * Microsoft.CodeAnalysis.CSharp.Analyzer.Testing.NUnit
-  * Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.NUnit
-  * Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing.NUnit
-  * Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing.NUnit
-* Visual Basic
-  * Microsoft.CodeAnalysis.VisualBasic.Analyzer.Testing.NUnit
-  * Microsoft.CodeAnalysis.VisualBasic.CodeFix.Testing.NUnit
-  * Microsoft.CodeAnalysis.VisualBasic.CodeRefactoring.Testing.NUnit
-  * Microsoft.CodeAnalysis.VisualBasic.SourceGenerators.Testing.NUnit
+This collection of packages previously included a number of packages for testing with specific test frameworks, such as
+MSTest, NUnit, and xUnit. These packages have been marked obsolete, and users are encouraged to migrate to the generic
+test packages. The migration process shown here is specific to MSTest; other frameworks can substitute `NUnit` or `XUnit`
+as appropriate.
 
-### xUnit.net
+* Remove the **.MSTest** suffix from all referenced packages. For example:
 
-* C#
-  * Microsoft.CodeAnalysis.CSharp.Analyzer.Testing.XUnit
-  * Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.XUnit
-  * Microsoft.CodeAnalysis.CSharp.CodeRefactoring.Testing.XUnit
-  * Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing.XUnit
-* Visual Basic
-  * Microsoft.CodeAnalysis.VisualBasic.Analyzer.Testing.XUnit
-  * Microsoft.CodeAnalysis.VisualBasic.CodeFix.Testing.XUnit
-  * Microsoft.CodeAnalysis.VisualBasic.CodeRefactoring.Testing.XUnit
-  * Microsoft.CodeAnalysis.VisualBasic.SourceGenerators.Testing.XUnit
+    Microsoft.CodeAnalysis.CSharp.Analyzer.Testing.MSTest &rarr; Microsoft.CodeAnalysis.CSharp.Analyzer.Testing
+
+* Remove the **.MSTest** suffix from namespaces. For example:
+
+    ```patch
+    -using Microsoft.CodeAnalysis.CSharp.Testing.MSTest;
+    +using Microsoft.CodeAnalysis.CSharp.Testing;
+    ```
+
+* Use `CSharpAnalyzerVerifier<TAnalyzer, DefaultVerifier>` instead of `AnalyzerVerifier<TAnalyzer>`. A similar change
```

Comment:
Could we consider defining a `CSharpAnalyzerVerify<TAnalyzer> : CSharpAnalyzerVerifier<TAnalyzer, DefaultVerifier>` here, to make migration easier?

Reply:
I considered creating an `AnalyzerVerifier<TAnalyzer>` which would bind as soon as the namespace changed, but worried that having two types with the same name would be an even worse situation. Note that the obsolete type is not typically used in a large number of cases, because the default project template creates a set of verifier types derived from the obsolete type directly in the test project, so changing the base type in one location would automatically update all tests using it.

Reply:
Makes sense. Maybe we consider doing that in a future version.